OpenAI/Packages/com.openai.unity/Runtime/Threads/CreateRunRequest.cs

```diff
@@ -1,30 +1,181 @@
 // Licensed under the MIT License. See LICENSE in the project root for license information.
 
 using Newtonsoft.Json;
+using System;
 using System.Collections.Generic;
 using System.Linq;
 using UnityEngine.Scripting;
 
 namespace OpenAI.Threads
 {
+    /// <summary>
+    /// Create a run on a thread.
+    /// </summary>
     [Preserve]
     public sealed class CreateRunRequest
     {
+        /// <summary>
+        /// Constructor.
+        /// </summary>
+        /// <param name="assistantId">
+        /// The ID of the assistant used for execution of this run.
+        /// </param>
+        /// <param name="request"><see cref="CreateRunRequest"/>.</param>
         [Preserve]
         public CreateRunRequest(string assistantId, CreateRunRequest request)
-            : this(assistantId, request?.Model, request?.Instructions, request?.Tools, request?.Metadata, request?.Temperature)
+            : this(
+                assistantId,
+                request?.Model,
+                request?.Instructions,
+                request?.AdditionalInstructions,
+                request?.AdditionalMessages,
+                request?.Tools,
+                request?.Metadata,
+                request?.Temperature,
+                request?.TopP,
+                request?.Stream ?? false,
+                request?.MaxPromptTokens,
+                request?.MaxCompletionTokens,
+                request?.TruncationStrategy,
+                request?.ToolChoice as string ?? (string)request?.ToolChoice?.funcion?.name,
```

Comment:
@StephenHodgson with `object` instead of `dynamic` this line fails with:
```'object' does not contain a definition for 'funcion' and no accessible extension method 'funcion' accepting a first argument of type 'object' could be found (are you missing a using directive or an assembly reference?)```

Reply:
Minor: And just noticed it says funcion without a t 😄

Reply:
Yes I figured there's be some work to do after converting back to `object`. 

Great catch on typo.